File: triangles.java
// Time Complexity : O(N)
// Space Complexity : O(N)
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : None

public class triangles {
    public int minimumTotal(List<List<Integer>> triangle) {
        int[][] dp = new int[triangle.size()][triangle.size()];
		for(int i=0;i<triangle.size() ;i++)
        {
            dp[triangle.size()-1][i]=triangle.get(triangle.size()-1).get(i);   
        }
        
        for(int i=triangle.size()-2;i>=0;i--){
            for(int j = 0; j <= i; j++)
            {
        dp[i][j]= triangle.get(i).get(j) + Math.min(dp[i+1][j], dp[i+1][j+1]);
            }
        }
       return dp[0][0]; 
    }
}

File: Triangles.py
#120. Triangle
#Time complexity: O(n)
# Space complexity: O(1)

class Solution:
    def minimumTotal(self, triangle: List[List[int]]) -> int:
        m = len(triangle)
        for i in range(m-2, -1, -1):
            for j in range(0, i+1):
                triangle[i][j] = triangle[i][j]+(min(triangle[i+1][j],triangle[i+1][j+1]))
        return triangle[0][0]
